src/python_testing/TC_OCC_3_1.py

```diff
@@ -1,129 +1,160 @@
+        ]
+        return steps
+
+    def pics_TC_OCC_3_1(self) -> list[str]:
+        pics = [
+            "OCC.S",
+        ]
+        return pics
+
+    # Sends and out-of-band command to the all-clusters-app
+    def write_to_app_pipe(self, command):
+        with open(self.app_pipe, "w") as app_pipe:
+            app_pipe.write(command + "\n")
+        # Delay for pipe command to be processed (otherwise tests are flaky)
+        time.sleep(0.001)
+
+    @async_test_body
+    async def test_TC_OCC_3_1(self):
+        hold_time = 10  # 10 seconds for occupancy state hold time
+
+        # CI app pipe id creation
+        self.app_pipe = "/tmp/chip_all_clusters_fifo_"
+        self.is_ci = self.check_pics("PICS_SDK_CI_ONLY")
+        if self.is_ci:
+            app_pid = self.matter_test_config.app_pid
+            if app_pid == 0:
+                asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set.c")
+            self.app_pipe = self.app_pipe + str(app_pid)
+
+        self.step(1)  # Commissioning already done
+
+        self.step(2)
+
+        cluster = Clusters.OccupancySensing
+        attributes = cluster.Attributes
+        attribute_list = await self.read_occ_attribute_expect_success(attribute=attributes.AttributeList)
+
+        has_hold_time = attributes.HoldTime.attribute_id in attribute_list
+
+        if has_hold_time:
+            # write 10 as a HoldTime attribute
+            await self.write_single_attribute(cluster.Attributes.HoldTime(hold_time))
+        else:
+            logging.info("No HoldTime attribute supports. Will test only occupancy attribute triggering functionality")
+
+        self.step(3)
+        # check if Occupancy attribute is 0
+        occupancy_dut = await self.read_occ_attribute_expect_success(attribute=attributes.Occupancy)
+
+        # if occupancy is on, then wait until the sensor occupancy state is 0.
+        if occupancy_dut == 1:
+            # Don't trigger occupancy sensor to render occupancy attribute to 0
+            if has_hold_time:
+                time.sleep(hold_time + 2.0)  # add some extra 2 seconds to ensure hold time has passed.
+            else:  # a user wait until a sensor specific time to change occupancy attribute to 0.  This is the case where the sensor doesn't support HoldTime.
+                # CI call to trigger off
+                if self.is_ci:
+                    self.write_to_app_pipe('{"Name":"SetOccupancy", "EndpointId": 1, "Occupancy": 0}')
+                else:
+                    self.wait_for_user_input(
+                        prompt_msg="Type any letter and press ENTER after the sensor occupancy is detection ready state (occupancy attribute = 0)")
+
+        # check sensor occupancy state is 0 for the next test step
+        occupancy_dut = await self.read_occ_attribute_expect_success(attribute=attributes.Occupancy)
+        asserts.assert_equal(occupancy_dut, 0, "Occupancy attribute is still 1.")

```

Comment:
Let's revisit the test plan to test the occupancy subscription updates in 3.1, not 3.2 and keep 3.2 for timing attribute subscription reporting.

Reply:
Mostly so.